src/Discord.Net.Commands/Attributes/Preconditions/RequirePermission.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Discord.Commands.Attributes.Preconditions
+{
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
+    public class RequirePermission : RequireGuildAttribute
+    {
+        public GuildPermission? GuildPermission { get; set; }
+        public ChannelPermission? ChannelPermission { get; set; }
+
```

Comment:
It just seems odd to have constructors for one property or the other but not both. This is pretty insignificant though, so it's fine if it's left like this.